tls/s2n_kem.h

```diff
@@ -52,14 +54,48 @@ struct s2n_iana_to_kem {
     uint8_t kem_count;
 };
 
+struct s2n_kem_group {
+    const char *name;
+    uint16_t iana_id;
+    const struct s2n_ecc_named_curve *curve;
+    const struct s2n_kem *kem;
+};
+
+struct s2n_kem_group_params {
+    const struct s2n_kem_group *kem_group;
+    struct s2n_kem_params kem_params;
+    struct s2n_ecc_evp_params ecc_params;
+};
+
 #if !defined(S2N_NO_PQ)
+    extern const struct s2n_kem s2n_bike1_l1_r1;
+    extern const struct s2n_kem s2n_bike1_l1_r2;
+    extern const struct s2n_kem s2n_sike_p503_r1;
+    extern const struct s2n_kem s2n_sike_p434_r2;
+
+    #define S2N_NUM_SECP256R1_KEM_GROUPS 1
+    extern const struct s2n_kem_group s2n_secp256r1_sike_p434_r2;
+
+    #if EVP_APIS_SUPPORTED
+        /* If EVP_APIS_SUPPORTED, then we support x25519 based kem_groups */
+        #define S2N_NUM_X25519_KEM_GROUPS 1
+        extern const struct s2n_kem_group s2n_x25519_sike_p434_r2;
+    #else
+        /* If *not* EVP_APIS_SUPPORTED, then we *do not* supported x25519 */
+        #define S2N_NUM_X25519_KEM_GROUPS 0
+    #endif
+
+    #define S2N_SUPPORTED_KEM_GROUPS_COUNT (S2N_NUM_SECP256R1_KEM_GROUPS + S2N_NUM_X25519_KEM_GROUPS)
 
-extern const struct s2n_kem s2n_bike1_l1_r1;
-extern const struct s2n_kem s2n_bike1_l1_r2;
-extern const struct s2n_kem s2n_sike_p503_r1;
-extern const struct s2n_kem s2n_sike_p434_r2;
+#else
+    /* If S2N_NO_PQ is defined, we must still define
+     * S2N_SUPPORTED_KEM_GROUPS_COUNT to be non-zero.
+     * This value is used to declare the size of
+     * certain arrays in s2n_crypto.h and ISO C forbids
+     * zero-size arrays. */
 
-#endif
+    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 1
```

Comment:
I'll note that I really don't like what I'm doing here to define `S2N_SUPPORTED_KEM_GROUPS_COUNT`.
* It's confusing `#ifdef` shenanigans
* If the build does NOT support PQ, we still define `S2N_SUPPORTED_KEM_GROUPS_COUNT` equal to 1, which is contradictory, confusing, and error prone.